### ISSUE TITLE
[Classic] prefer Light/Dark Theme setting

### ISSUE THREAD

**Issue by c33s** [MrAlex94/Waterfox#1980]
any chance to add `prefers-color-scheme` to waterfox classic?

https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme

**Comment by jabcreations** [comment]
I just commented on this thread:
https://github.com/MrAlex94/Waterfox/issues/1996#issuecomment-803464044

**Comment by hawkeye116477** [comment]
Yes, I'm working on it. Linux part done, testing now, will check W10 soon too.

![2021-03-25_11-53](https://user-images.githubusercontent.com/19818572/112461906-59e8ce80-8d58-11eb-8cd4-75da780c7caa.png)

**Comment by jabcreations** [comment]
Absolutely wild! Please feel free to test dark mode on my site for further validation: https://www.jabcreations.com/
Thank you for your efforts! 😍︀😍︀😍︀

**Comment by reallyuniquename** [comment]
@hawkeye116477
Would `ui.systemUsesDarkTheme` config entry be supported as well?

**Comment by hawkeye116477** [comment]
@reallyuniquename Yes.

**Comment by jabcreations** [comment]
I just updated to Waterfox 56 2021.03 and dark mode isn't working. Additionally I tested window.matchMedia('(prefers-color-scheme: dark)').matches and it returns false. What is going on?

**Comment by hawkeye116477** [comment]
@jabcreations Preference `privacy.resistFingerprinting` must be set to `false` to make it working.

**Comment by jabcreations** [comment]
@hawkeye116477 It _is_ set to `false`. Anything else?

**Comment by hawkeye116477** [comment]
@jabcreations What's your OS?

**Comment by jabcreations** [comment]
Windows 10 Pro x64.

**Comment by hawkeye116477** [comment]
So maybe you got wrong build. I saw that few days ago was hidden 2021.03 installer and that was build before adding prefers-color-scheme feature. Go to `about:buildconfig` and check if you have `Built from https://github.com/MrAlex94/Waterfox/commit/5d6e36f2f62aee766e62e19be21793334ed7c9c7`. If not, then try to download installer again. Otherwise maybe try in safe mode/new profile.

**Comment by jabcreations** [comment]
I also tested a new profile earlier and forgot to mention that it still didn't work. I just restarted with addons disabled and it still doesn't work. Here is the full page dump of about:buildconfig, hopefully this helps:

```
about:buildconfig
Source

Built from https://github.com/MrAlex94/Waterfox/commit/5d6e36f2f62aee766e62e19be21793334ed7c9c7
Build platform
target
x86_64-pc-mingw32
Build tools
Compiler 	Version 	Compiler flags
c:/Users/akontos/.cargo/bin/sccache.exe clang-cl.exe -Xclang -std=gnu99 -fms-compatibility-version=19.11.25547 	19.11.25547 	-Qunused-arguments -TC -nologo -wd4091 -D_HAS_EXCEPTIONS=0 -W3 -Gy -Zc:inline -Gw -wd4244 -wd4267 -Wno-unknown-pragmas -Wno-ignored-pragmas -Wno-deprecated-declarations -Wno-invalid-noreturn -we4553
c:/Users/akontos/.cargo/bin/sccache.exe clang-cl.exe -fms-compatibility-version=19.11.25547 	19.11.25547 	-Qunused-arguments -Qunused-arguments -TP -nologo -wd5026 -wd5027 -Zc:sizedDealloc- -wd4091 -wd4577 -D_HAS_EXCEPTIONS=0 -W3 -Gy -Zc:inline -Gw -wd4251 -wd4244 -wd4267 -wd4800 -wd4595 -Wno-inline-new-delete -Wno-invalid-offsetof -Wno-microsoft-enum-value -Wno-microsoft-include -Wno-unknown-pragmas -Wno-ignored-pragmas -Wno-deprecated-declarations -Wno-invalid-noreturn -Wno-inconsistent-missing-override -Wno-implicit-exception-spec-mismatch -Wno-unused-local-typedef -Wno-ignored-attributes -we4553 -GR- -Zi -O2 -Qvec -w -clang:-arch:SSE3 -Oy-
c:/Users/akontos/.cargo/bin/rustc.exe 	1.49.0 	
Configure options

'MOZILLABUILD=C:\mozilla-build\' --host=x86_64-pc-mingw32 --target=x86_64-pc-mingw32 --enable-application=browser --enable-update-channel=release --disable-tests '--enable-optimize=-O2 -Qvec -w -clang:-arch:SSE3' NASM=c:/Users/akontos/.mozbuild/nasm/nasm --enable-ccache=sccache --with-visual-studio-version=2017 CC=clang-cl CXX=clang-cl --enable-release --enable-rust-simd MAKE=c:/mozilla-build/bin/mozmake.EXE --disable-crashreporter --disable-maintenance-service --disable-signmar --enable-updater --disable-verify-mar --with-app-basename=Waterfox --with-app-name=waterfox --with-branding=browser/branding/unofficial --with-distribution-id=org.waterfoxproject
```

**Comment by hawkeye116477** [comment]
Well, so it looks like correct build. You can also check checksum of installer, SHA256 should be `bf41539d1c2178143f8456ff29feb1c7179213f931a90aedda1f50984ca0a444`. You can also go to `about:config` and create `ui.systemUsesDarkTheme` integer preference and set it to `1` and check if it's working.

Can I also ask what's your Windows 10 version and build (https://www.howtogeek.com/236195/how-to-find-out-which-build-and-version-of-windows-10-you-have/)?

**Comment by jabcreations** [comment]
Creating `ui.systemUsesDarkTheme` and setting the integer to 1 did not correct the problem.

How do I find the hash generated by the install?

Windows 10 Pro x64 Version 1909 Build 18,363.418 twice removed on my Uncle's side in order for me to test IE18 (I'm the lead developer for web platform) before they replaced the rendering engine with yet another useless Chrome clone. It most certainly supports dark mode.

`window.matchMedia('(prefers-color-scheme: dark)')`

generates:

`MediaQueryList { media: "(prefers-color-scheme: dark)", matches: false, onchange: null }`

**Comment by hawkeye116477** [comment]
> Creating ui.systemUsesDarkTheme and setting the integer to 1 did not correct the problem.

Ok, so if that didn't cause to change mode to dark, then probably it's some problem on your side. It should work even if OS didn't support dark mode.

I have almost same Windows 10 1909 build 18363.959.

> How do I find the hash generated by the install?

For example install https://github.com/namazso/OpenHashTab/releases/tag/v3.0.1 and then right click on installer file and choose properties and hashes tab.

**Comment by jabcreations** [comment]
Okay I tried the OpenHashTab program and even logged off and then restated and no context menu or Hashs tab in the properties window for the Waterfox installer I downloaded earlier today appeared. Any other suggestions?

**Comment by hawkeye116477** [comment]
> no context menu or Hashs tab in the properties window for the Waterfox installer

Very strange, it works fine for me. Maybe your AV did joke, some of them could report false positive. If you have another language, then `Hashs` tab can be also translated, but I suppose you figured it out.

I can also suggest https://github.com/idrassi/HashCheck/releases, installer is signed, so shouldn't be problematic :smile:

**Comment by jabcreations** [comment]
Okay, that second _checksum_ suggestion actually worked!

bf41539d1c2178143f8456ff29feb1c7179213f931a90aedda1f50984ca0a444 *Waterfox Classic 2021.03 Setup.exe

While I wait for what I hope is another suggestion (I am _not_ trying to burn through them!) I'll temporarily move all of the installed files and do a "fresh" install and test with a fresh profile again. If you really want I can probably upload the ISO (I presume I have it) if you really want to install it in VMware Player or something and see if you can see what the heck is going on.

**Comment by hawkeye116477** [comment]
Well, another suggestion is that maybe you should try updating OS or installing OS for fresh. 

If I would guess what you did, then I would said that maybe you used NTLite or WinReducer and you removed too much from W10 installer (or you even got ready "slim" installer for W10), so that's why you have that problems now. There is also possibility that you used CCleaner or similar tool. Avast or another AV could also sometimes made bad joke.

**Comment by jabcreations** [comment]
I sincerely apologize, I just went to my site this morning and dark mode was working. The zoom feature was no longer working on any website (I use 2560x1440 on my 4K screen because 32 inches isn't sufficient for that many pixels) and `privacy.resistFingerprinting` was set to `true` which is odd because you mentioned to set it to `false`! Well I set it to `false` (again, apparently). I set Windows 10 to light mode and it didn't go back to light mode on my site so I discovered `ui.systemUsesDarkTheme` set to `1` will override the OS apparently. I'm not sure how we got everything confused though I'll add some documentation on the Fixed Firefox I maintain to make it easier for folks. Thank you again for your hard work!
